docs/zeebe/dynamic_scaling.md

```diff
@@ -0,0 +1,209 @@
+   "replicationFactor": 3
+   }
+   }' | jq
+   ```
+
+   - without brokers
+
+   ```bash
+   curl -X 'PATCH' \
+   'http://localhost:9600/actuator/cluster' \
+   -H 'accept: application/json' \
+   -H 'Content-Type: application/json' \
+   -d '{  
+   "brokers": { 
+   "add": [3,4,5]
+   },
+   "partitions": {
+   "count": 6, 
+   "replicationFactor": 3
+   }
+   }' | 
+   ```
+2. The [ClusterChangeCoordinator](../../zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/changes/ConfigurationChangeCoordinator.java) generates the `ClusterChangeOperation` needed to reach the requested result.
+   For a pure scaling operation, the generated `ClusterChangeOperation` will be:
+   - `StartPartitionScaleUp` to Partition 1
+   - `PartitionBootstrapOperation` to all new partitions
+   - `AwaitRedistributionCompletion` to the coordinator node 1
+
+   Once generated, the coordinator node will execute one operation at a time, on the correct node.
+
+   **Partition ID Requirements**: New partitions must have contiguous IDs (e.g., if you currently have partitions 1-3, you can only scale to 4, 5, 6, etc.). Non-contiguous partition IDs are rejected to maintain compatibility with existing routing logic.
+
+3. `StartPartitionScaleUp`:
+
+   - Partition 1 marks that the scale up is started in the state.
+   - new partitions are added to the `desiredPartitionCount` in the state
+   - commands distribution will enqueue new commands for the new partitions, but it will not start distributing them yet.
+4. `PartitionBootstrapOperation` on the new partitions (one partition at a time, from the *lowest* to the *highest*):
+   - The partition is bootstrapped in the node with the flag `bootstrapFromSnapshot=true`.
+   - It will request the snapshot for bootstrap to partition 1.
+   - When the snapshot is received, it will restore the `PersistedSnapshotStore` with it, before the partition is even started (in a `PartitionStartupStep`)
+   - When the bootstrap is completed, it will inform partition 1 that they are ready to start new process instances
+   - Partition 1 will include the newly bootstrapped partitions to the `activePartitions` field in `RoutingInfo` in the state
+   - Partition 1 will start distributing enqueue distribution commands to the new partition
+   - Partition 1 will distribute to all partitions the newly received bootstrap command
+   - All other partition will do the same changes to the `RoutingInfo` in the state
+
+   **Bootstrap Validation**: Each partition validates the received snapshot before marking itself as bootstrapped. If validation fails, the operation is retried automatically.
+
+5. `AwaitRedistributionCompletion`:
+   `CoordinatorNode` will keep polling Partition 1 to know when all partitions have been bootstrapped.
+
+   When redistribution is completed, the routing information in the dynamic config module will update, so that gateways will deploy new process instances to the newly bootstrapped partitions as well.
```

Comment:
:question: Is the routing state in dynamic config only updated after all partitions are bootstrapped?